server/controllers/Users.js

```diff
@@ -0,0 +1,47 @@
+import models from '../database/models';
+// import { generateToken, serverResponse, serverError } from '../helpers';
+import { serverResponse, serverError } from '../helpers';
+import hashPassword from '../helpers/passwords';
+
+const { User } = models;
+
+/**
+ * @export
+ * @class User
+ */
+class Users {
+  /**
+   * @name signUp
+   * @async
+   * @static
+   * @memberof User
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new user
+   */
+  static async signUp(req, res) {
+    try {
+      const { email, userName, password } = req.body;
+      const existingEmail = await User.findOne({ where: { email } });
+      const existingUserName = await User.findOne({ where: { userName } });
+      if (existingEmail) {
+        return serverResponse(res, 409, { error: 'Email already exists' });
+      }
+      if (existingUserName) {
+        return serverResponse(res, 409, { error: 'Username already exists' });
+      }
+      const hashedPassword = await hashPassword(password);
+      const user = await User.create({
+        ...req.body,
+        password: hashedPassword
+      });
+      delete user.dataValues.password;
+      // const token = await generateToken(user.id);
+      return serverResponse(res, 201, { user });
+    } catch (error) {
+      return serverError(res);
```

Comment:
Nice work. I like the fact that you also cater for error handling. However, do you think the server error response should be called directly in every controller method? Why don't we have an express error handler middleware in our server entry file which is in place to respond also to the errors that may occur even when controller method connection is closed.